homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,283 @@
+      station_id: 11035
+      monitored_conditions:
+        - temperature
+        - humidity
+        - pressure
+        - wind_speed
+        - precipitation
+
+Recognised conditions are:
+
+    pressure (Pressure at station level)
+    pressure_sealevel (Pressure at Sea Level)
+    humidity (Humidity)
+    wind_speed (Wind Speed)
+    wind_bearing (Wind Bearing)
+    wind_max_speed (Top Wind Speed)
+    wind_max_bearing (Top Wind Bearing)
+    sun_last_hour (Sun Last Hour Percentage)
+    temperature (Temperature)
+    precipitation (Precipitation)
+    dewpoint (Dew Point)
+
+The following stations are available:
+
+    11010   Linz/Hörsching
+    11012   Kremsmünster
+    11022   Retz
+    11035   Wien/Hohe Warte
+    11036   Wien/Schwechat
+    11101   Bregenz
+    11121   Innsbruck
+    11126   Patscherkofel
+    11130   Kufstein
+    11150   Salzburg
+    11155   Feuerkogel
+    11157   Aigen im Ennstal
+    11171   Mariazell
+    11190   Eisenstadt
+    11204   Lienz
+"""
+
+import csv
+from datetime import timedelta
+import logging
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    WeatherEntity, PLATFORM_SCHEMA, ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_PRESSURE, ATTR_WEATHER_TEMPERATURE,
+    ATTR_WEATHER_WIND_BEARING, ATTR_WEATHER_WIND_SPEED,
+    ATTR_WEATHER_ATTRIBUTION
+)
+from homeassistant.const import (
+    TEMP_CELSIUS, CONF_MONITORED_CONDITIONS,
+    CONF_NAME, STATE_UNKNOWN
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'zamg'
+ATTRIBUTION = 'Data provided by ZAMG'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)  # Only updates once per hour
+
+CONF_STATION_ID = "station_id"
+
+SENSOR_TYPES = {
+    ATTR_WEATHER_PRESSURE: ('Pressure', 'hPa', 'LDstat hPa', float),
+    'pressure_sealevel': ('Pressure at Sea Level', 'hPa', 'LDred hPa', float),
+    ATTR_WEATHER_HUMIDITY: ('Humidity', '%', 'RF %', int),
+    ATTR_WEATHER_WIND_SPEED: ('Wind Speed', 'km/h', 'WG km/h', float),
+    ATTR_WEATHER_WIND_BEARING: ('Wind Bearing', '°', 'WR °', int),
+    'wind_max_speed': ('Top Wind Speed', 'km/h', 'WSG km/h', float),
+    'wind_max_bearing': ('Top Wind Bearing', '°', 'WSR °', int),
+    'sun_last_hour': ('Sun Last Hour', '%', 'SO %', int),
+    ATTR_WEATHER_TEMPERATURE: ('Temperature', '°C', 'T °C', float),
+    'precipitation': ('Precipitation', 'l/m²', 'N l/m²', int),
+    'dewpoint': ('Dew Point', '°C', 'TP °C', float),
+    # The following probably not useful for general consumption,
+    # but we need them to fill in internal attributes
+    'station_name': ('Station Name', None, 'Name', str),
+    'station_elevation': ('Station Elevation', 'm', 'Höhe m', int),
+    'update_date': ('Update Date', None, 'Datum', str),
+    'update_time': ('Update Time', None, 'Zeit', str),
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.string,
```

Comment:
You specify a list of available stations however you do not validate that the given station id is one of them.